x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -55,15 +77,84 @@ func (k msgServer) SubmitProof(ctx context.Context, msg *types.MsgSubmitProof) (
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// Unmarshal the closest merkle proof from the message.
+	sparseMerkleClosestProof := &smt.SparseMerkleClosestProof{}
+	if err := sparseMerkleClosestProof.Unmarshal(msg.GetProof()); err != nil {
+		return nil, types.ErrProofInvalidProof.Wrapf(
+			"failed to unmarshal closest merkle proof: %s",
+			err,
+		)
+	}
+
+	// Get the relay request and response from the proof.GetClosestMerkleProof.
+	closestValueHash := sparseMerkleClosestProof.ClosestValueHash
+	relayBz := closestValueHash[:len(closestValueHash)-sumSize]
+	relay := &servicetypes.Relay{}
+	if err := k.cdc.Unmarshal(relayBz, relay); err != nil {
+		return nil, types.ErrProofInvalidRelay.Wrapf(
+			"failed to unmarshal relay: %s",
+			err,
+		)
+	}
+
+	if err := relay.GetReq().ValidateBasic(); err != nil {
```

Comment:
@red-0ne WHat's the thinking/plan around testing each of these?

I realize we'll have an E2E test that covers one happy path but there might be an opportunity for a few simple unit tests. 

1. Open to push back. Lmk
2. If there's pushback. I'd personally copy-paste this file into ChatGPT and have it do the first ~30% of the work for me (if you prompt it correctly)
3. TODO_TEST is recommended if we decide to do this.

Reply:
We definitely need to have tests for the req/res `ValidateBasic` functions, and these function more love.

I'll opt for adding `TODO_TEST` above their declarations now and test them in another PR.